src/tests/suite/daos_obj.c

```diff
@@ -989,22 +983,44 @@ io_rewritten_array_with_mixed_size(void **state)
 			OW_IOD_SIZE, size, DAOS_TX_NONE, &req);
 		assert_memory_equal(ow_buf, fbuf, size);
 
-		/*Verify the pool size*/
-		rc = pool_storage_info(state, &pinfo);
-		assert_int_equal(rc, 0);
-		nvme_current_size = pinfo.pi_space.ps_space.s_free[1];
 		/**
-		*Data written on SCM so NVMe free size should not change.
-		*/
-		if (nvme_current_size != nvme_initial_size) {
-			fail_msg("NVMe_current_size =%"
-				PRIu64" != NVMe_initial_size %" PRIu64"",
-				nvme_current_size, nvme_initial_size);
+		 * Data written on SCM so NVMe free size should not change.
+		 * However VEA free (called from aggregation) usually put the
+		 * freed extent in an aging buffer for 10 seconds
+		 * so this while loop will check the size every 2 seconds
+		 * for a 30 seconds.
+		 */
+		tmp = 1;
+		while (tmp <= 15) {
+			/*Verify the pool size*/
+			rc = pool_storage_info(state, &pinfo);
+			assert_int_equal(rc, 0);
+			nvme_current_size = pinfo.pi_space.ps_space.s_free[1];
+
+			if (nvme_current_size == nvme_initial_size)
+				break;
+
+			/**
+			* Test will fail if it has been tried for
+			* 15 times (30 seconds) and NVMe size is not
+			* the same as initial size.
+			*/
+			if ((nvme_current_size != nvme_initial_size) &&
+			    tmp == 20) {
+				fail_msg("NVMe_current_size =%"
+					PRIu64" != NVMe_initial_size %"
+					PRIu64" for record set=%d",
+					nvme_current_size, nvme_initial_size,
+					record_set);
+			}
+			sleep(2);
+			tmp++;
```

Comment:
hmm, shouldn't the test goes like:
1. get initial nvme free size;
2. write to SCM;
3. get current nvme free, if "current nvme free == initial nvme free", test passed, otherwise; 
4. if total sleep time > timeout, test failed, otherwise;
5. sleep 2 seconds and go back to 1st step;

I don't quite follow your above changes.

Reply:
Below is the line of functionality 

> 1. get initial nvme free size; --> On line 968
> 2. write to SCM; --> On line 978
> 3. get current nvme free, if "current nvme free == initial nvme free", test passed, otherwise; --> On line 1000
> 4. if total sleep time > timeout, test failed, otherwise; --> On line 1008/1009 where size is not matched and its loop for 20 times (30 seconds) will fail the test.
> 5. sleep 2 seconds and go back to 1st step;
> 
> I don't quite follow your above changes.

So you want to do SCM write again after 2 seconds if the size does not match?

Reply:
Yes, I think the point of this test is to verify if NVMe free size is affected by SCM update, am I understanding correct?

Reply:
Done, Updated code